### PR TITLE
Fixing NullReferenceException in XmlSchemaAnyAttribute.Namespace

### PR DESCRIPTION
**Pull request by martin-frydl**
Fix for issue #41704 with test,

### CODE REVIEW

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs [hunk 1]
```diff
@@ -240,6 +240,30 @@ public void v3(string ns, string attrNs, string expectedNs)
             CompareWildcardNamespaces(expectedNs, attributeWildcard.Namespace);
         }
 
+        [Fact]
+        //[Variation(Desc = "new instance return null namespace (issue #41704")]
+        public void v4()
```
Comment (bartonjs) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs]: The existing test names are from probably several test suites ago.  The preferred format now is to have the name convey the purpose of the test.  Then you can delete the commented out Variation attribute :smile:.

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs [hunk 2]
```diff
@@ -240,6 +240,30 @@ public void v3(string ns, string attrNs, string expectedNs)
...
+        //[Variation(Desc = "without namespace returns null (issue #41704"")]
+        public void v5()
+        {
+            var xsd = @"<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema' targetNamespace='ns'>
+                        <xs:complexType name = 't'>
+                            <xs:anyAttribute/>
+                        </xs:complexType>
+                        </xs:schema>";
+            var xs = XmlSchema.Read(new StringReader(xsd), null);
+            var any = ((XmlSchemaComplexType)xs.Items[0]).AnyAttribute;
```
Comment (bartonjs) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs]: Our style only allows `var` when the type is "obvious", which means a `new`, `(cast)`, or `as cast`.  All three of the `var`s in this method should be replaced with the actual variable types.

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs [hunk 3]
```diff
@@ -240,6 +240,30 @@ public void v3(string ns, string attrNs, string expectedNs)
             CompareWildcardNamespaces(expectedNs, attributeWildcard.Namespace);
         }
 
+        [Fact]
+        //[Variation(Desc = "new instance return null namespace (issue #41704")]
+        public void v4()
+        {
+            var any = new XmlSchemaAnyAttribute();
+            Assert.Null(any.Namespace);
+            return;
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs]: Nit: delete this `return;` line

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs [hunk 4]
```diff
@@ -240,6 +240,30 @@ public void v3(string ns, string attrNs, string expectedNs)
...
+        {
+            var xsd = @"<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema' targetNamespace='ns'>
+                        <xs:complexType name = 't'>
+                            <xs:anyAttribute/>
+                        </xs:complexType>
+                        </xs:schema>";
+            var xs = XmlSchema.Read(new StringReader(xsd), null);
+            var any = ((XmlSchemaComplexType)xs.Items[0]).AnyAttribute;
+            Assert.Null(any.Namespace);
+            return;
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs]: Nit: delete this `return;` line

**Review by stephentoub — COMMENTED** [review]
The actual fix LGTM.

File: src/System.Private.Xml/src/System/Xml/Schema/XmlSchemaAnyAttribute.cs [hunk 5]
```diff
@@ -17,7 +17,7 @@ public class XmlSchemaAnyAttribute : XmlSchemaAnnotated
         [XmlAttribute("namespace")]
         public string Namespace
         {
-            get { return _ns ?? NamespaceList.ToString(); }
+            get { return _ns ?? NamespaceList?.ToString(); }
```
Comment (krwq) [inline src/System.Private.Xml/src/System/Xml/Schema/XmlSchemaAnyAttribute.cs]: was this returning null or empty string before?

Comment (martin-frydl) [reply src/System.Private.Xml/src/System/Xml/Schema/XmlSchemaAnyAttribute.cs]: It was returning null.

**Comment by martin-frydl** [comment]
The tests are failing but I don't think that is caused by my code:
.packages/microsoft.dotnet.helix.sdk/5.0.0-beta.19510.4/tools/azure-pipelines/AzurePipelines.MultiQueue.targets(16,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Request to https://dev.azure.com/dnceng/public/_apis/test/runs?api-version=5.0-preview.2 returned failed status 503 Service Unavailable

**Comment by danmosemsft** [comment]
Infra was down. trying again.

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs [hunk 6]
```diff
@@ -240,6 +240,26 @@ public void v3(string ns, string attrNs, string expectedNs)
...
+        public void NewInstanceReturnsNullNamespace()
+        {
+            var any = new XmlSchemaAnyAttribute();
+            Assert.Null(any.Namespace);
+        }
+
+        [Fact]
+        public void ReadFromFileWithoutNamespaceReturnsNull()
+        {
+            var xsd = @"<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema' targetNamespace='ns'>
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_AnyAttribute.cs]: Nit: var => string

**Review by stephentoub — APPROVED** [review]

**Review by krwq — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**